pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -725,6 +726,34 @@ private static String getAuthenticationMethodName(int authReq) {
     }
   }
 
+  private static final int AUTH_METHOD_NONE = 1;
+  private static final int AUTH_METHOD_PASSWORD = 2;
+  private static final int AUTH_METHOD_MD5 = 4;
+  private static final int AUTH_METHOD_GSS = 8;
+  private static final int AUTH_METHOD_SSPI = 16;
+  private static final int AUTH_METHOD_SCRAM_SHA_256 = 32;
+
+  private static int getAuthMethodBit(String method) {
+    switch (method) {
+      case "none": return AUTH_METHOD_NONE;
+      case "password": return AUTH_METHOD_PASSWORD;
+      case "md5": return AUTH_METHOD_MD5;
+      case "gss": return AUTH_METHOD_GSS;
+      case "sspi": return AUTH_METHOD_SSPI;
+      case "scram-sha-256": return AUTH_METHOD_SCRAM_SHA_256;
+      default: return 0;
+    }
+  }
+
+  private static void validateAuthMethod(int methodBit, int allowedMethods, int rejectedMethods) throws PSQLException {
+    if ((rejectedMethods & methodBit) != 0) {
+      throw new PSQLException(GT.tr("Authentication method is rejected by requireAuth"), PSQLState.CONNECTION_REJECTED);
+    }
+    if (allowedMethods != 0 && (allowedMethods & methodBit) == 0) {
+      throw new PSQLException(GT.tr("Authentication method is not allowed by requireAuth"), PSQLState.CONNECTION_REJECTED);
+    }
+  }
+
```

Comment:
Why are we doing it with bit masks rather than throwing the allowed / disallowed in a `Set<string>(...)` and calling `.has(...)`?

We have to do the mapping from the FEBE code to a name once and then can use that for the check. It's not like this happening in some tight loop. It'd be a single check for the whole connection handshake right?